File: codegen/results/v002/m3-max-48gb/ollama/qwen3.5_4b-nvfp4_nothink/20260412_082024/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strings"
)

func main() {
	file, err := os.Open("input/text.txt")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	wordFreq := make(map[string]int)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		// Create regex to keep only letters (a-z), ignoring case and handling punctuation
		pattern := regexp.MustCompile(`[^a-zA-Z]`)
		words := pattern.Split(line, -1)

		for _, word := range words {
			if len(word) > 0 {
				lowerWord := strings.ToLower(word)
				wordFreq[lowerWord]++
			}
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	var words []string
	for word := range wordFreq {
		words = append(words, word)
	}

	// Sort by count descending, then alphabetically ascending
	sort.Slice(words, func(i, j int) bool {
		if wordFreq[words[i]] != wordFreq[words[j]] {
			return wordFreq[words[i]] > wordFreq[words[j]]
		}
		return words[i] < words[j]
	})

	output := make([]byte, 0)
	for _, word := range words {
		fmt.Printf("%s: %d\n", word, wordFreq[word])
	}
}

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.5_4b-nvfp4_nothink/20260412_082024/python/002_word_frequency/solution.py
import string
from collections import Counter

def process_file(filename):
    try:
        with open(filename, 'r', encoding='utf-8') as f:
            text = f.read()
    except FileNotFoundError:
        print("File not found.")
        return

    # Remove punctuation and split into words
    # Keep only letters (a-z) and strip to get words
    translator = str.maketrans('', '', string.punctuation)
    cleaned_text = text.lower().translate(translator)

    # Split by whitespace to get words
    words = cleaned_text.split()

    # Filter for actual words (non-empty after split, though split handles multiple spaces)
    words = [word for word in words if word]  # Ensure non-empty strings

    # Count frequencies
    word_counts = Counter(words)

    # Sort by count descending, then alphabetically ascending for ties
    sorted_words = sorted(word_counts.keys(), key=lambda w: (-word_counts[w], w.lower()))

    # Output results
    for word in sorted_words:
        print(f"{word}: {word_counts[word]}")

if __name__ == "__main__":
    process_file("input/text.txt")